content/blog/2023-03-18-monitor-kubernetes-network-by-ebpf/kubernetes-service-list.md

```diff
@@ -0,0 +1,199 @@
+Given the necessity to filter process traffic based on certain criteria such as the process's namespace,
+Apache SkyWalking Rover follows a series of steps to ensure accurate and efficient monitoring. These steps include:
+
+1. **Monitoring Activation**: The process is immediately added to a monitoring whitelist upon detection.
+   This step ensures that the process is considered for monitoring from the moment it starts, without delay.
+2. **Push to Queue**: The process's PID (Process ID) is pushed into a monitoring confirmation queue.
+   This queue holds the PIDs of newly detected processes that are pending further confirmation from a user-space program.
+   This asynchronous approach allows for the separation of immediate detection and subsequent processing, optimizing the monitoring workflow.
+3. **User-Space Program Confirmation**: The user-space program retrieves process PIDs from the queue and assesses whether
+   each process should continue to be monitored. If a process is deemed unnecessary for monitoring, it is removed from the whitelist.
+
+This process ensures that SkyWalking Rover can dynamically adapt its monitoring scope based on real-time conditions and configurations, allowing for both comprehensive coverage and efficient resource use.
+
+#### Limitations
+
+The monitoring of stateful protocols like HTTP/2.x currently faces certain limitations:
+
+1. **Inability to Observe Pre-existing Connections**: Monitoring the complete request and response cycle requires that monitoring be initiated before any connections are established.
+   This requirement means that connections set up before the start of monitoring cannot be observed.
+2. **Challenges with TLS Requests**: Observing TLS encrypted traffic is complex because it relies on **asynchronously attaching uprobes** (user-space attaching) for observation.
+   If new requests are made before these uprobes are successfully attached, it becomes impossible to access the data before encryption or after decryption.
+
+## Demo
+
+Next, let’s quickly demonstrate the Kubernetes monitoring feature, so you can understand more specifically what it accomplishes.
+
+### Deploy SkyWalking Showcase
+
+SkyWalking Showcase contains a complete set of example services and can be monitored using SkyWalking. For more information, please check the [official documentation](https://skywalking.apache.org/docs/skywalking-showcase/next/readme/).
+
+In this demo, we only deploy service, the latest released SkyWalking OAP, and UI.
+
+```shell
+export SW_OAP_IMAGE=ghcr.io/apache/skywalking/oap
+export SW_OAP_IMAGE_TAG=f4836327a65efe76cbbc12f013cc9cbc63a82b2c
+export SW_UI_IMAGE=ghcr.io/apache/skywalking/ui
+export SW_UI_IMAGE_TAG=f4836327a65efe76cbbc12f013cc9cbc63a82b2c
+export SW_ROVER_IMAGE=ghcr.io/apache/skywalking-rover/skywalking-rover:d3a0f874642cf00a5e252440ee3b76a1cea3aac4
+
+export FEATURE_FLAGS=java-agent-injector,single-node,elasticsearch, rover
+make deploy.kubernetes
+```
```

Comment:
> Should we wait for the rover and OAP release steps, and then publish this blog?

I think once you update the showcase to the latest, we don't need to export these, and the current and showcase in the future are both working. 
So, we dont' have to wait, right?

Reply:
Showcase is using the dev versions, rather than released versions.